### PR TITLE
added cmsearch-deoverlap tool

### PR DESCRIPTION
**Pull request by RZ9082**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection

cmsearch-deoverlap removes overlapping hits to models in the same clan in a single --tblout  cmsearch output file.

### CODE REVIEW

File: tools/cmsearch_deoverlap/cmsearch-deoverlap.xml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    <tests>
+        <test>
+            <param name="clan_information" value="ribo.claninfo" />
+            <param name="cmsearch_matches" value="1.cmsearch.tblout" />
+            <output name="deoverlapped_matches" file="1.cmsearch.tblout.deoverlapped">
+            </output>
+        </test>
+    </tests>
+    <help><![CDATA[
+Usage:
```
Comment (paulzierep) [inline tools/cmsearch_deoverlap/cmsearch-deoverlap.xml:29]: I think the part can be removed.

File: tools/cmsearch_deoverlap/cmsearch-deoverlap.xml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    </requirements>
+    <command><![CDATA[
+        ln -s '$clan_information' clan_info;
+        ln -s '$cmsearch_matches' cmsearch_matches;
+        LC_ALL=C perl '$__tool_directory__/cmsearch-deoverlap.pl' --clanin clan_info cmsearch_matches;
+        mv *.deoverlapped '$deoverlapped_matches'
+
+    ]]></command>
+    <inputs>
+        <param type="data" name="clan_information" format="txt" label="clan information"/>
```
Comment (paulzierep) [inline tools/cmsearch_deoverlap/cmsearch-deoverlap.xml:14]: A small help text would be good.

**Review by paulzierep — COMMENTED** [review]
some text changes

File: tools/cmsearch_deoverlap/cmsearch-deoverlap.xml [hunk 3]
```diff
@@ -0,0 +1,60 @@
+<tool id="cmsearch_deoverlap" name="CMsearch deoverlap" version="0.08+galaxy0">
```
Comment (bgruening) [inline tools/cmsearch_deoverlap/cmsearch-deoverlap.xml:1]: please add a profile="21.05"

File: tools/cmsearch_deoverlap/cmsearch-deoverlap.xml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    <description>removes lower scoring overlaps from cmsearch --tblout files</description>
```
Comment (bgruening) [inline tools/cmsearch_deoverlap/cmsearch-deoverlap.xml:2]:
```suggestion
    <description>removes lower scoring overlaps from cmsearch results</description>
```

**Review by bgruening — COMMENTED** [review]
@RZ9082 @paulzierep this script has no parameter and is very specific to one exact tool/outputs. Should we add this as option to cmsearch directly?

File: tools/cmsearch_deoverlap/cmsearch-deoverlap.pl [hunk 5]
```diff
@@ -0,0 +1,495 @@
...
+# hits 1 and 4 no longer overlap.
+#
+use strict;
+use warnings;
+use Getopt::Long;
+use File::Basename;
+
+my $in_tblout  = "";   # name of input tblout file
+
+my $usage;
```
Comment (paulzierep) [inline tools/cmsearch_deoverlap/cmsearch-deoverlap.pl:41]: Can we add the sort hit (bit score / E-value), clanin or not, and maxkeep / not max keep as parameter ?

**Review by paulzierep — COMMENTED** [review]
add parameters

**Comment by paulzierep** [comment]
> @RZ9082 @paulzierep this script has no parameter and is very specific to one exact tool/outputs. Should we add this as option to cmsearch directly?

I think there are 3 parameters that should be added, see my review.
I had similar thoughts about integrating it in cmsearch, but there could be cases where one would want to use it on old data, and therefore I would prefer to have it separately.

**Comment by paulzierep** [comment]
> Please use latest version from original repo
> 
> https://github.com/nawrockie/cmsearch_tblout_deoverlap/blob/master/cmsearch-deoverlap.pl
> 
> Please ask there if they can add a license otherwise we can not distribute it: https://choosealicense.com/no-permission/
> 
> May fit better to @bgruenings rnateam repo

Interesting, if EBI-Metagenomics copied it into their repo: https://github.com/EBI-Metagenomics/pipeline-v5 and added a license, we cannot use it since the original does not have one, right ?

**Comment by bernt-matthias** [comment]
> we cannot use it since the original does not have one, right ?

No. Just ask the maintainers of the original repo to add a license.

**Pull request closed without merging**